sway-core/src/ir_generation/function.rs

```diff
@@ -1735,16 +1804,26 @@ impl<'eng> FnCompiler<'eng> {
         // cache, to uniquely identify a function instance, is the span and the type IDs of any
         // args and type parameters.  It's using the Sway types rather than IR types, which would
         // be more accurate but also more fiddly.
-        let fn_key = (
-            callee.span(),
-            callee
-                .parameters
-                .iter()
-                .map(|p| p.type_argument.type_id)
-                .collect(),
-            callee.type_parameters.iter().map(|tp| tp.type_id).collect(),
-        );
-        let new_callee = match self.recreated_fns.get(&fn_key).copied() {
+
+        let (fn_key, item) = if !callee.span().as_str().is_empty() {
```

Comment:
Here is how I understand the change done to function generation: if the function is not compiler generated (`!callee.span().as_str().is_empty()`) generate the key and either get already compiled function by that key or create it and insert under that key. But if it is compiler generated, generate it every time it is called.

I guess this works for now because the functions generated for contracts calls, and also `__entry` are called only once. Still, looking into future and possible generation of functions for different purposes (thinking of storage right now), can we think of:
1. Adding `is_compiler_generated()` method to `TyFunctionDecl` and eventually think of other ways of determined if a function is compiler generated. E.g. in the _compiler_generated.rs_ we used name prefixes to determine if variables are compiler generated.
2. Create a unique `fn_key` also for the compiler generated methods. I assume we will still have enough distinguished elements like e.g. unique names and the parameters and type parameters we will anyhow have.

Reply:
I will answer you in the comments because I think this is important.